components/datetime/src/parts.rs

```diff
@@ -0,0 +1,118 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! Parts of a formatted date/time.
+//!
+//! # Examples
+//!
+/// ```
+/// use icu::calendar::Gregorian;
+/// use icu::calendar::{Date, Time};
+/// use icu::datetime::DateTimeWriteError;
+/// use icu::datetime::parts;
+/// use icu::datetime::fieldsets;
+/// use icu::datetime::DateTimeFormatter;
+/// use icu::locale::locale;
+/// use icu::timezone::IxdtfParser;
+/// use writeable::assert_writeable_parts_eq;
+///
+/// let dtf = DateTimeFormatter::try_new(
+///     locale!("en-u-ca-buddhist").into(),
+///     fieldsets::YMDTZ::medium(),
+/// )
+/// .unwrap();
+///
+/// let dtz = IxdtfParser::new().try_from_str("2023-11-20T11:35:03+00:00[Europe/London]").unwrap();
+///
+/// // Missing data is filled in on a best-effort basis, and an error is signaled.
+/// assert_writeable_parts_eq!(
```

Comment:
suggestion: can we actually show how parts-writing can be used here?  If I find this module I won't know how to use it, the assertion does not show or link to off the underlying API.

Reply:
Yeah, so there isn't currently a "nice" way to use these; you need to write your own implementation of PartsWrite. The naive-always-allocating implementation used in the macro is not intended to be public.

Reply:
Ah, okay. That's fair, feel free to merge. 

I think we might want better signposting in the assert_writeable docs so that people going there from a docstest know what to look for. I'll file a followup.